queue_job_cron/models/ir_cron.py

```diff
@@ -32,14 +32,17 @@ def _run_job_as_queue_job(self, server_action):
         return server_action.run()
 
     def method_direct_trigger(self):
-        if self.run_as_queue_job:
-            return self.with_delay(
-                priority=self.priority,
-                description=self.name,
-                channel=self.channel_id.complete_name,
-            )._run_job_as_queue_job(server_action=self.ir_actions_server_id)
-        else:
-            return super().method_direct_trigger()
+        for cron in self:
+            if not cron.run_as_queue_job:
+                super(IrCron, cron).method_direct_trigger()
+            else:
+                _cron = cron.with_user(cron.user_id).with_context(lastcall=cron.lastcall)
+                _cron.with_delay(
+                    priority=self.priority,
+                    description=self.name,
```

Comment:
but then... all the `self` here should be replaced w/ `cron` :wink:

Reply:
tip: next time delegate this to a method `_run_as_queue_job()` and forget about this mess

Reply:
Yes. You see the quick hack nature of this.